### ISSUE TITLE
node.removeSelf() and reattaching it is not possible

### ISSUE THREAD

**Issue by MoOx** [postcss/postcss#196]
I guess we should add one more test somewhere before doing this ones :)

```
/.../postcss/import/node_modules/postcss/lib/node.js:361
            if (p != root && p.parent && p.parent == root) {
                              ^
TypeError: Cannot read property 'parent' of undefined
    at /.../postcss/import/node_modules/postcss/lib/node.js:361:31
    at /.../postcss/import/node_modules/postcss/lib/container.js:155:24
    at Root.each (/.../postcss/import/node_modules/postcss/lib/container.js:128:20)
    at Root.eachInside (/.../postcss/import/node_modules/postcss/lib/container.js:154:21)
    at Rule.style (/.../postcss/import/node_modules/postcss/lib/node.js:359:16)
    at Rule.style (/.../postcss/import/node_modules/postcss/lib/node.js:439:31)
    at Rule.stringifyBlock (/.../postcss/import/node_modules/postcss/lib/container.js:65:27)
    at Rule.stringify (/.../postcss/import/node_modules/postcss/lib/rule.js:77:14)
    at Root.stringifyContent (/.../postcss/import/node_modules/postcss/lib/container.js:53:25)
    at Root.stringify (/.../postcss/import/node_modules/postcss/lib/root.js:109:14)
```

Ref https://github.com/postcss/postcss-import/issues/21

**Comment by ai** [comment]
I can add a test, but i don’t know how to write it :).

**Comment by ai** [comment]
Fixed 1392efe

**Comment by ai** [comment]
Released in 4.0.4.
